util/pkg/vfs/context.go

```diff
@@ -348,3 +355,27 @@ func (c *VFSContext) buildOpenstackSwiftPath(p string) (*SwiftPath, error) {
 
 	return NewSwiftPath(c.swiftClient, bucket, u.Path)
 }
+
+func (c *VFSContext) buildOSSPath(p string) (*OSSPath, error) {
+	u, err := url.Parse(p)
+	if err != nil {
+		return nil, fmt.Errorf("invalid aliyun oss path: %q", p)
+	}
+
+	if u.Scheme != "oss" {
+		return nil, fmt.Errorf("invalid aliyun oss path: %q", p)
+	}
+
+	bucket := strings.TrimSuffix(u.Host, "/")
+	if bucket == "" {
+		return nil, fmt.Errorf("invalid aliyun oss path: %q", p)
+	}
+
+	ossClient, err := NewAliOSSClient()
+	if err != nil {
+		return nil, err
+	}
+	c.ossClient = ossClient
```

Comment:
You may want to cache this (if there's some state), it looks like right now we put the value into a field but don't really use it (technically there's a potential threading race)

Reply:
Thanks, I've fixed it by reusing `c.ossClient` when it's not nil.